### ISSUE TITLE
Logger Tags don't get broadcasted to another Tagged Logger

### ISSUE THREAD

**Issue by jvillarejo** [rails/rails#43291]
I'm using `ActiveSupport::Logger.broadcast` to broadcast logging to multiple loggers. 
For example STDOUT, File in `log/` folder and another `RemoteSysLogger` 

Also I'm using `ActiveSupport::TaggedLogging` to configure tags with `middleware` on the config file:

```ruby
  config.log_tags = [:host, :request_id] 
```

But this log tags don't get broadcasted to the other logs. 
I don't know if there something that I'm configuring wrong.

### Steps to reproduce
Add on `development.rb` 

```ruby
  config.log_tags = [:host, :uuid]
  config.after_initialize do
    other_logger = ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new(Rails.root.join('log', "other.log")))
    Rails.logger.extend(ActiveSupport::Logger.broadcast(other_logger))
  end
```

Make a GET request to root_path '/' 

### Expected behavior
I expected that `other.log` file had request tags: [:host, :uuid]

```
# log/development.log
[localhost] [cfb7f921-0a7c-476a-8b0c-166963b87e4f] Started GET "/" for ::1 at 2021-09-22 15:44:50 -0300
```

```
# log/other.log
[localhost] [cfb7f921-0a7c-476a-8b0c-166963b87e4f] Started GET "/" for ::1 at 2021-09-22 15:44:50 -0300
```

### Actual behavior
However other log didn't include tags

```
# log/development.log
[localhost] [cfb7f921-0a7c-476a-8b0c-166963b87e4f] Started GET "/" for ::1 at 2021-09-22 15:44:50 -0300
```

```
# log/other.log
Started GET "/" for ::1 at 2021-09-22 15:44:50 -0300
```

### System configuration
**Rails version**: 6.1
**Ruby version**: 2.7

**Comment by jvillarejo** [comment]
In order to make it work I apply a monkey patch on my application:

```ruby

module TaggedBroadcast
  extend ActiveSupport::Concern

  class_methods do
    def broadcast(logger)
      super(logger).tap do |broadcast_logger_class|
        broadcast_logger_class.define_method(:tagged) do |*tags, &block|
          if logger.respond_to?(:tagged)
            logger.tagged(*tags) do 
              if defined?(super)
                super(*tags, &block)
              else
                block.call(self)
              end
            end
          else
            if defined?(super)
              super(*tags, &block)
            else
              block.call(self)
            end
          end
        end
      end
    end
  end
end

ActiveSupport::Logger.prepend(TaggedBroadcast)
```

I can submit a PR for this fix in `ActiveSupport:Logger` so it support `tagged` if the logger `responds_to?(:tagged)` 
Or if you give me a better directions on how to solve it, I can do it too. :smile:

**Comment by rafaelfranca** [comment]
We have a test to make sure this works https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/activesupport/test/tagged_logging_test.rb#L221, and this test is passing.

I think you need to investigate why your setup is not behaving in the same way as the test.

**Comment by rafaelfranca** [comment]
Ah, it seems it is only present in Rails 7. https://github.com/rails/rails/commit/70af536b5d6e57800cb6b6b9a10f33b39192113e. I'll backport to 6-1-stable.

Fixed by b21394cd20

**Comment by jvillarejo** [comment]
Cool, thanks! 
Sorry I didn't search before posting :crying_cat_face:
